bin/peekaboo-util.py

```diff
@@ -43,49 +90,27 @@ def main():
                         help='Output additional diagnostics')
     parser.add_argument('-s', '--socket_file', action='store', required=True,
                         help='Path to Peekaboo\'s socket file')
-    parser.add_argument('-f', '--filename', action='append', required=True,
-                        help='Path to the file to scan. Can be given more '
-                        'than once to scan multiple files.')
-    args = parser.parse_args()
-
-    result_regex = re.compile(r'.*wurde als',
-                              re.MULTILINE + re.DOTALL + re.UNICODE)
-    peekaboo = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
-    peekaboo.connect(args.socket_file)
 
-    file_snippets = []
-    for filename in args.filename:
-        file_snippets.append('{ "full_name": "%s" }' % path.abspath(filename))
-    request = '[ %s ]' % ', '.join(file_snippets)
+    scan_file_parser = subparsers.add_parser('scan_file',
+                                             help='Scan a file and report it')
+    scan_file_parser.add_argument('-f', '--filename', action='append', required=True,
```

Comment:
Just a remark as heads-up: The required option sequence is peekaboo-util.py -s &lt;sockfile&gt; scan_file -f &lt;file&gt;. When putting -s after the command name, an error is printed "error: argument -s/--socket_file is required" which can be confusing. From my point of view it's certainly consistent with the command and parser structure but dunno how easy we want to make it for users. We *could* add global arguments to each subcommand as well. No idea if it's worth the effort.

Reply:
The help states nicely where to put the arguments.
I will test in the next workshop how people react to it.

```
$ bin/peekaboo-util.py
usage: peekaboo-util.py [-h] [-v] [-vv] [-d] -s SOCKET_FILE {scan-file} ...
peekaboo-util.py: error: too few arguments
$ bin/peekaboo-util.py scan-file -h
usage: peekaboo-util.py scan-file [-h] -f FILENAME
```

Reply:
Agreed.